### DISCUSSION TITLE
redis.clients.jedis.exceptions.JedisConnectionException: java.net.SocketTimeoutException: Read timed out

### DISCUSSION THREAD

**Discussion by anil181** [xetorthio/jedis#2571] [Q&A] [answered]
**jedisClient.mset()** is throwing redis.clients.jedis.exceptions.JedisConnectionException: java.net.SocketTimeoutException: Read timed out.
<img width="501" alt="timeout" src="https://user-images.githubusercontent.com/7187168/123595647-c6b7ad00-d80e-11eb-9668-7ff924985c76.PNG">

Am using jedis 3.6.0 with Redis OSS 6.2.3.

All timeouts set to 60 seconds for sentinelConfig and masterConfig:
connectionTimeoutMillis(60000).blockingSocketTimeoutMillis(60000).blockingSocketTimeoutMillis(60000).build();

**Comment by yangbodong22011** [comment, marked as answer]
Timeout is usually related to many factors. It is recommended that you follow the steps below to troubleshoot:

### 1 About Redis
1.1 Is Redis CPU too high, or whether the machine where Redis is loaded is too high?
1.2 Is Redis executing some O(n) commands when this exception is obtained, such as keys, hgetall, etc.

### 2 About Jedis

2.1 Are Jedis parameters configured correctly?
2.2 Whether the load of Jedis machine is too high

###  3. About the network
3.1 Whether there is a problem with the network from Jedis to Redis machine (p.s. If a timeout occurs by accident, it is likely to be caused by the network)

**Reply by anil181** [reply]
Thank you for the prompt reply. Issue was coming as the SocketTomeout was not setup.

**Comment by sazzad16** [comment]
Besides what @yangbodong22011 said, check if the request (`mset()`) is too large.

**Reply by anil181** [reply]
Thank you for the prompt reply. Issue was coming as the SocketTomeout was not setup.
